packages/e2e-test/src/utils.ts

```diff
@@ -0,0 +1,315 @@
+import {
+  HexString,
+  Address,
+  utils,
+  Cell,
+  Script,
+  TransactionWithStatus,
+  Block,
+} from "@ckb-lumos/base";
+import { Indexer, CellCollector } from "@ckb-lumos/ckb-indexer";
+import {
+  CKBIndexerQueryOptions,
+  OtherQueryOptions,
+} from "@ckb-lumos/ckb-indexer/lib/type";
+import { common, dao } from "@ckb-lumos/common-scripts";
+import {
+  TransactionSkeleton,
+  encodeToAddress,
+  parseAddress,
+  sealTransaction,
+} from "@ckb-lumos/helpers";
+import { key, mnemonic, ExtendedPrivateKey, AddressType } from "@ckb-lumos/hd";
+import {
+  getConfig,
+  ScriptConfigs,
+  initializeConfig,
+  predefined,
+  createConfig,
+  Config,
+} from "@ckb-lumos/config-manager";
+import { RPC } from "@ckb-lumos/rpc";
+import { BI } from "@ckb-lumos/bi";
+import { INDEXER_RPC_URI, CKB_RPC_URI } from "./constants";
+
+export const defaultIndexer = new Indexer(INDEXER_RPC_URI, CKB_RPC_URI);
+export const defaultRPC = new RPC(CKB_RPC_URI);
+
+export async function getGenesisSciprtConfig(
+  options: { rpc?: RPC } = {}
+): Promise<ScriptConfigs> {
+  const { rpc: _rpc = defaultRPC } = options;
+  const genesisBlock = await _rpc.getBlockByNumber("0x0");
+
+  const secp256k1DepTxHash = genesisBlock.transactions[1].hash;
+  const secp256k1TypeScript = genesisBlock.transactions[0].outputs[1].type;
+  const secp256k1TypeHash = utils.computeScriptHash(secp256k1TypeScript!);
+
+  const daoDepTxHash = genesisBlock.transactions[0].hash;
+  const daoTypeScript = genesisBlock.transactions[0].outputs[2].type;
+  const daoTypeHash = utils.computeScriptHash(daoTypeScript!);
+
+  return {
+    SECP256K1_BLAKE160: {
+      HASH_TYPE: "type",
+      CODE_HASH: secp256k1TypeHash,
+      TX_HASH: secp256k1DepTxHash!,
+      INDEX: "0x0",
+      DEP_TYPE: "depGroup",
+    },
+    DAO: {
+      HASH_TYPE: "type",
+      CODE_HASH: daoTypeHash,
+      TX_HASH: daoDepTxHash!,
+      INDEX: "0x2",
+      DEP_TYPE: "code",
+    },
+  };
+}
+
+export const loadConfig = async (): Promise<Config> => {
+  const _genesisConfig = await getGenesisSciprtConfig();
+
+  const CONFIG = createConfig({
+    PREFIX: "ckt",
+    SCRIPTS: {
+      ...predefined.AGGRON4.SCRIPTS,
+      ..._genesisConfig,
+    },
+  });
+
+  initializeConfig(CONFIG);
+  return CONFIG;
+};
+
+export const generateFirstHDPrivateKey = (): string => {
+  const myMnemonic = mnemonic.generateMnemonic();
+  const seed = mnemonic.mnemonicToSeedSync(myMnemonic);
+
+  const extendedPrivKey = ExtendedPrivateKey.fromSeed(seed);
+  return extendedPrivKey.privateKeyInfo(AddressType.Receiving, 0).privateKey;
+};
+
+export function asyncSleep(ms: number): Promise<unknown> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export async function waitTransactionCommitted(
```

Comment:
i found most utility functions in `utils.ts` are dependent on rpc, indexer, pollInterval, maybe we should encapsulate them for more to make it easier to call

```ts
interface E2EProvider {
  readonly pollIntervalMs: number; 
  readonly indexer: Indexer;
  readonly rpc: RPC; 

  // send a transaction to claim CKB
  claimCkb(options: { to: LockScriptLike, amount: BIish }): Promise<Hash>;
  
  // wait for transaction status to be committed
  waitTransactionCommitted(options: { txHash: Hash, timeout: number }): Promise<void>;

  // wait for the block height to greater than or equal to a given value
  // for ckb + indexer
  waitForBlock(options: { relative: boolean, value: number }): Promise<void>;

  // wait for the epoch to greater than or equal to a given value
  // for ckb + indexer
  waitForEpoch(options: {relative: boolean, value: number }): Promise<void>
}
```